### ISSUE TITLE
Two failing tests.

### ISSUE THREAD

**Issue by eloisabentivegna** [Sbozzolo/kuibit#20]
Tests `test_antenna_responses` and `test_ra_dec_to_theta_phi` are failing for me at the moment, with errors:

AssertionError: 0.1676658308656924 != 0.173418558 within 7 places (0.005752727134307606 difference)

and

AssertionError: 2.458839344228527 != 2.3697174 within 7 places (0.08912194422852693 difference)

respectively.

Could this be the effect of any of the changes implemented as a result of the review in https://github.com/openjournals/joss-reviews/issues/3099?

**Comment by Sbozzolo** [comment]
That's quite interesting. All the tests are running fine on my computers and the supercomputers I have access to, and on GitHub actions.

![2021-04-10:08:41:21](https://user-images.githubusercontent.com/9167485/114275786-ad098500-99d8-11eb-8c5a-9473325b01fa.png)

The failure of the second test leads to the failure of the first. So, I have to understand why the second fails. The only possible reason I can think is that the `datetime` Python module is behaving differently on your system. Other than that, the function only does numerical manipulations of numbers with `numpy` and it is very difficult to see how they could lead to different results.

Can you add the following print statement after line 290 of `gw_utils.py` and run the test again?
```python
print(base_posix_date, posix_date)
```
On my computer, it prints `946753200.0 1442249445.0`.

Are you running the test in the poetry virtual environment? What version of Python and OS are you using?

**Comment by eloisabentivegna** [comment]
Sure thing. I get

> .............................................................946728000.0 1442220645.0
> 946728000.0 1442220645.0
> ..........................................................................946728000.0 1442220645.0
> 946728000.0 1442220645.0
> 946728000.0 1442220645.0
> 946728000.0 1442220645.0
> .946728000.0 1442220645.0
> 946728000.0 1442220645.0
> 946728000.0 1442220645.0
> 946728000.0 1442220645.0
> ...946728000.0 1442220645.0
> F..946728000.0 1442220645.0
> F..............................................................
> ======================================================================
> FAIL: test_antenna_responses (tests.test_gw_utils.TestGWUtils)
> ----------------------------------------------------------------------

I am using Python 3.8 on OS 10.15.

**Comment by Sbozzolo** [comment]
Yes, they are different! This might be related to some behavior specific to macOS and how it defines local times. 

Can you please try to change the two lines that define `base_posix_data` and `posix_date` to
```python
    base_posix_date = datetime.datetime.strptime(
        "2000-01-01 12:00:00 UTC", "%Y-%m-%d %H:%M:%S %Z"
    ).timestamp()
    posix_date = datetime.datetime.strptime(
        time_utc + " UTC", "%Y-%m-%d %H:%M:%S %Z"
    ).timestamp()
```
I am forcing the timezone to be UTC. Hopefully this works.

**Comment by Sbozzolo** [comment]
I added macOS to the CI and the tests are all passing even without that change. There must be a subtle problem, but I hope that the proposed change will fix it.

**Comment by eloisabentivegna** [comment]
The change does not make a difference (which is at least consistent with your second message). There's probably a version mismatch in some of the dependencies. Given the magnitude of the difference, I am not considering this a blocker for the JOSS review -- test maintenance is a continuing process for evolving software anyway.

Regardless of the review, I am happy to send more info on the other components of my setup, if this can help. Or you can tell me which system configuration you used on macOS and I can figure out where the differences are.

**Comment by Sbozzolo** [comment]
> There's probably a version mismatch in some of the dependencies

If you are running with `poetry`, it should use the very same versions I am using (the ones defined in the `poetry.lock`), so I don't think that's the problem. I would point my finger to some difference between macOS and GNU/Linux, but simply changing the GitHub Action runner to macOS didn't reveal any issue. 

>  Or you can tell me which system configuration you used on macOS and I can figure out where the differences are.

I have no idea. I don't have access to any computer with macOS (at least, in an easy way). My test on macOS is with the configurations provided by [this](https://github.com/actions/setup-python), with `poetry`.


Does the program
```python
import datetime
datetime.datetime.strptime( "2000-01-01 12:00:00", "%Y-%m-%d %H:%M:%S").timestamp()
```
return `946753200.0`?

**Comment by eloisabentivegna** [comment]
Nope, `946728000.0`.

**Comment by Sbozzolo** [comment]
Ok, I can reproduce the problem and it has to do with the local time on the machine. I am not sure what is exactly the problem, but I figure it out.

**Comment by eloisabentivegna** [comment]
Makes sense. It's probably a locale-related thing -- the time I get from the command above is exactly seven hours behind yours. I can't exclude it's something messed up with my system (although it would be best to find a way to prevent it from influencing the test results).

**Comment by Sbozzolo** [comment]
Yeah, what I don't understand is that the timezone shouldn't matter for the computation because I take a difference of two times, so the common offset should cancel out. Anyways, I changed the way I do that computation (https://github.com/Sbozzolo/kuibit/commit/7ad44235af4b4a3044d8fb1e5bf7fbfd2810822b).  I tested the code on the machine on a different timezone and the test is passing, so I think that this has been solved.
